### PR TITLE
perf: zero copy send

### PR DESCRIPTION
**Pull request by qzhuyan**
- zero copy send
-  fix some binary leakage
- Stream immediate shutdown when stream owner is dead

### CODE REVIEW

File: test/quicer_snb_SUITE.erl [hunk 1]
```diff
@@ -230,6 +231,85 @@ tc_slow_conn(Config) ->
                end),
   ok.
 
+tc_stream_owner_down(Config) ->
```
Comment (qzhuyan) [inline test/quicer_snb_SUITE.erl:234]: Ease to combine and check with three players in one test, I fall in love with `snk`, thank you! @k32

File: c_src/quicer_connection.c [hunk 2]
```diff
@@ -130,6 +130,7 @@ _IRQL_requires_max_(DISPATCH_LEVEL)
   QuicerStreamCTX *s_ctx = NULL;
   ErlNifEnv *env = c_ctx->env;
   ERL_NIF_TERM report;
+  BOOLEAN is_destroy = FALSE;
```
Comment (k32) [inline c_src/quicer_connection.c:133]: `is_destroyed`, perhaps ?

File: test/quicer_snb_SUITE.erl [hunk 3]
```diff
@@ -230,6 +231,85 @@ tc_slow_conn(Config) ->
...
+                                             Trace)),
+
+                   %% check that client side immediate shutdown trigger an peer_send_abort event at server side
+                   ?assert(?strict_causality(#{ ?snk_kind := debug
+                                              , context := "callback"
+                                              , function := "ClientStreamCallback"
+                                              , tag := "event"
+                                              , mark := 6
+                                              },
+                                             #{'$kind' := peer_send_aborted
```
Comment (k32) [inline test/quicer_snb_SUITE.erl:305]: '$kind' -> `?snk_kind`

File: test/quicer_snb_SUITE.erl [hunk 4]
```diff
@@ -230,6 +231,85 @@ tc_slow_conn(Config) ->
...
+                                              },
+                                             #{ ?snk_kind := debug
+                                              , function := "resource_stream_down_callback"
+                                              , tag := "start"
+                                              , mark := 0
+                                              , resource_id := _RidS
+                                              },
+                                             _RidS == _RidL,
+                                             Trace)),
+                   %% check that it triggered a immediate stream shutdown
```
Comment (k32) [inline test/quicer_snb_SUITE.erl:281]: Variables bound in the first match pattern are in the scope of the second match pattern, so this statement could be simplified like this:

```suggestion
                   ?assert(?strict_causality(#{ ?snk_kind := debug
                                              , function := "stream_controlling_process"
                                              , tag := "exit"
                                              , resource_id := _Rid
                                              },
                                             #{ ?snk_kind := debug
                                              , function := "resource_stream_down_callback"
                                              , tag := "start"
                                              , mark := 0
                                              , resource_id := _Rid                                              },
                                             Trace)),
                   %% check that it triggered a immediate stream shutdown
```

**Review by k32 — APPROVED** [review]

**Pull request merged**